--- PKGBUILD-bk	2017-02-17 05:38:47.000000000 +0300
+++ PKGBUILD	2017-02-17 05:39:37.136338320 +0300
@@ -36,10 +36,8 @@
   cd $pkgname
 
   ./configure --prefix=/usr --sysconfdir=/etc --localstatedir=/var --libexecdir=/usr/lib \
-    --with-package-name="GStreamer VAAPI Plugin (Arch Linux)" \
-    --with-package-origin="http://www.archlinux.org/" \
     --disable-builtin-libvpx \
-    --enable-experimental --enable-gtk-doc --disable-static
+    --enable-experimental --disable-gtk-doc --disable-static
 
   # https://bugzilla.gnome.org/show_bug.cgi?id=655517
   sed -i -e 's/ -shared / -Wl,-O1,--as-needed\0/g' libtool
@@ -47,11 +45,6 @@
   make
 }
 
-check() {
-  cd $pkgname
-  make check
-}
-
 package() {
   cd $pkgname
   make DESTDIR="$pkgdir" install
