--- PKGBUILD.orig	2017-01-11 15:49:17.000000000 +0300
+++ PKGBUILD	2017-01-25 14:08:45.763452074 +0300
@@ -1,6 +1,6 @@
 # Contributor: Zeph <zeph33@gmail.com>
 # Maintainer: Zeph <zeph33@gmail.com>
-pkgname=pamac-aur
+pkgname=pamac
 pkgver=4.2.0
 _pkgver=4.2.0
 pkgrel=1
@@ -9,7 +9,7 @@
 url="https://github.com/manjaro/pamac"
 license=('GPL3')
 depends=('glib2>=2.42' 'json-glib' 'libsoup' 'dbus-glib' 'polkit' 'vte3>=0.38' 'gtk3>=3.18'
-         'libnotify' 'desktop-file-utils' 'pacman>=5.0' 'gnutls>=3.4')
+         'desktop-file-utils' 'pacman>=5.0' 'gnutls>=3.4')
   optdepends=('polkit-gnome: needed for authentification in Cinnamon, Gnome'
               'lxsession: needed for authentification in Xfce, LXDE etc.'
               'yaourt: needed for AUR support'
@@ -29,6 +29,7 @@
   cd "$srcdir/pamac-$_pkgver"
   sed -i -e "s|\"$_pkgver\"|\"$pkgver-$pkgrel\"|g" src/transaction.vala
   # patches here
+  patch -p0 -i ../../../no-notify.patch
 }
 
 build() {
