--- PKGBUILD	2016-11-28 21:20:37.000000000 +0300
+++ PKGBUILD	2016-11-28 21:20:13.432614303 +0300
@@ -9,13 +9,12 @@
 url="https://gstreamer.freedesktop.org/"
 arch=(i686 x86_64)
 license=(LGPL)
-depends=(mjpegtools gst-plugins-base-libs curl chromaprint libmms faad2 celt libdca libdvdnav
-         libmodplug libgme wayland libofa openjpeg2 libwebp libsrtp gnutls glu sbc rtmpdump
-         libgudev graphene)
-makedepends=(schroedinger libexif libdvdread libvdpau libmpeg2 python valgrind wildmidi librsvg
-             gobject-introspection gtk-doc gtk3 clutter libtiger ladspa openal libusb
-             qt5-declarative qt5-x11extras qt5-wayland vulkan-headers vulkan-icd-loader zbar
-             autoconf-archive git libfdk-aac faac soundtouch spandsp fluidsynth neon)
+depends=(gst-plugins-base-libs curl libmms faad2
+         libmodplug openjpeg2 libwebp libsrtp gnutls glu rtmpdump)
+makedepends=(libexif libvdpau python valgrind librsvg
+             gobject-introspection gtk-doc gtk3 ladspa openal libusb
+             vulkan-headers vulkan-icd-loader
+             autoconf-archive git faac neon)
 options=(!emptydirs)
 _commit=1d058f8ef55c6fb7011d84329f4c592d41aae35c  # tags/1.10.1^0
 source=("git://anongit.freedesktop.org/gstreamer/gst-plugins-bad#commit=$_commit"
@@ -40,6 +39,7 @@
 
 build() {
   cd $pkgname
+echo "#define VK_KHR_WAYLAND_SURFACE_EXTENSION_NAME \"VK_KHR_wayland_surface\"" >> ext/vulkan/vkdisplay.h
 
   ./configure --prefix=/usr --sysconfdir=/etc --localstatedir=/var --libexecdir=/usr/lib \
     --with-package-name="GStreamer Bad Plugins (Arch Linux)" \
