--- PKGBUILD.orig	2017-01-12 11:54:16.000000000 +0300
+++ PKGBUILD	2017-01-12 11:56:57.687518767 +0300
@@ -13,10 +13,10 @@
 license=('GPL')
 url='http://mpv.io'
 depends=(
-  'ffmpeg' 'lcms2' 'libcdio-paranoia' 'libgl' 'libxss'
-  'libxinerama' 'libxv' 'libxkbcommon' 'libva' 'wayland' 'libcaca'
-  'desktop-file-utils' 'hicolor-icon-theme' 'xdg-utils' 'lua52' 'libdvdnav'
-  'libxrandr' 'jack' 'smbclient' 'rubberband' 'uchardet'
+  'ffmpeg' 'lcms2' 'libgl' 'libxss'
+  'libxinerama' 'libxv' 'libxkbcommon' 'libva'
+  'desktop-file-utils' 'hicolor-icon-theme' 'xdg-utils'
+  'libxrandr' 'libass'
 )
 makedepends=('mesa' 'python-docutils' 'ladspa' 'hardening-wrapper')
 optdepends=('youtube-dl: for video-sharing websites playback')
@@ -37,8 +37,8 @@
 
   ./waf configure --prefix=/usr \
     --confdir=/etc/mpv \
-    --enable-cdda \
-    --enable-encoding \
+    --disable-cdda \
+    --disable-encoding \
     --enable-libmpv-shared \
     --enable-zsh-comp
 
