--- PKGBUILD-bk	2017-01-12 16:51:59.000000000 +0300
+++ PKGBUILD	2017-01-12 16:53:45.163963975 +0300
@@ -36,17 +36,6 @@
             '9765bca5d63fb5525bbd0520b7ab1d27cabaed697e2fc7791400abc3fa4f13b8')
 validpgpkeys=('2B90598A745E992F315E22C58AB132963A06537A')
 
-# Google API keys (see http://www.chromium.org/developers/how-tos/api-keys)
-# Note: These are for Arch Linux use ONLY. For your own distribution, please
-# get your own set of keys. Feel free to contact foutrelis@archlinux.org for
-# more information.
-_google_api_key=AIzaSyDwr302FpOSkGRpLlUpPThNTDPbXcIn_FM
-
-# Mozilla API keys (see https://location.services.mozilla.com/api)
-# Note: These are for Arch Linux use ONLY. For your own distribution, please
-# get your own set of keys. Feel free to contact heftig@archlinux.org for
-# more information.
-_mozilla_api_key=16674381-f021-49de-8622-3021c5942aff
 
 
 prepare() {
@@ -61,12 +50,6 @@
   # Build with the rust targets we actually ship
   patch -Np1 -i ../rust-i686.patch
 
-  echo -n "$_google_api_key" >google-api-key
-  echo "ac_add_options --with-google-api-keyfile=\"$PWD/google-api-key\"" >>.mozconfig
-
-  echo -n "$_mozilla_api_key" >mozilla-api-key
-  echo "ac_add_options --with-mozilla-api-keyfile=\"$PWD/mozilla-api-key\"" >>.mozconfig
-
   mkdir "$srcdir/path"
   ln -s /usr/bin/python2 "$srcdir/path/python"
 }
