--- PKGBUILD.orig	2017-03-04 00:48:43.000000000 +0300
+++ PKGBUILD	2017-03-04 00:51:44.244685127 +0300
@@ -12,9 +12,8 @@
 url=http://darktable.sf.net/
 license=('GPL3')
 depends=('exiv2>=0.18' 'lcms2' 'lensfun>=0.2.3' 'desktop-file-utils'
-	 'curl' 'libsecret' 'libgphoto2' 'openexr' 'sqlite' 'libxslt'
-	 'libsoup' 'gtk3' 'pugixml' 'json-glib' 'flickcurl' 'lua52'
-	 'colord' 'colord-gtk' 'graphicsmagick' 'osm-gps-map')
+	 'curl' 'libsecret' 'openexr' 'sqlite' 'libxslt'
+	 'libsoup' 'gtk3' 'pugixml' 'json-glib' 'osm-gps-map')
 makedepends=('intltool>=0.40' 'cmake' 'librsvg' 'osm-gps-map' 'libcups')
 optdepends=('librsvg' 'libcups')
 validpgpkeys=('C4CBC150699956E2A3268EF5BB5CC8295B1779C9')
@@ -35,9 +34,9 @@
       -DBINARY_PACKAGE_BUILD=1 \
       -DBUILD_USERMANUAL=False \
       -DUSE_LIBSECRET=On \
-      -DUSE_LUA=On \
+      -DUSE_LUA=Off \
       -DUSE_GNOME_KEYRING=Off \
-      -DUSE_COLORD=On \
+      -DUSE_COLORD=Off \
       -DPROJECT_VERSION=$pkgver \
       ..
   make
