diff -Naur cnchi-bk/scripts/postinstall.sh cnchi/scripts/postinstall.sh
--- cnchi-bk/scripts/postinstall.sh	2017-02-01 18:36:26.000000000 +0000
+++ cnchi/scripts/postinstall.sh	2017-02-01 20:10:04.941013211 +0000
@@ -55,15 +55,15 @@
 	if [[ "${CN_KEYBOARD_LAYOUT}" != '' ]]; then
 	  if [[ "${CN_KEYBOARD_VARIANT}" != '' ]]; then
 		  sed -i "s/'us'/'${CN_KEYBOARD_LAYOUT}+${CN_KEYBOARD_VARIANT}'/" /usr/share/cnchi/scripts/set-settings
-		  sed -i "s/'us'/'${CN_KEYBOARD_LAYOUT}+${CN_KEYBOARD_VARIANT}'/" "${CN_SCHEMA_OVERRIDE}"
+		  #sed -i "s/'us'/'${CN_KEYBOARD_LAYOUT}+${CN_KEYBOARD_VARIANT}'/" "${CN_SCHEMA_OVERRIDE}"
 	  else
 		  sed -i "s/'us'/'${CN_KEYBOARD_LAYOUT}'/" /usr/share/cnchi/scripts/set-settings
-		  sed -i "s/'us'/'${CN_KEYBOARD_LAYOUT}'/" "${CN_SCHEMA_OVERRIDE}"
+		  #sed -i "s/'us'/'${CN_KEYBOARD_LAYOUT}'/" "${CN_SCHEMA_OVERRIDE}"
 	  fi
 	fi
 
-	sed -i "s|@CN_BROWSER@|${CN_BROWSER}|g" "${CN_SCHEMA_OVERRIDE}"
-	cp "${CN_SCHEMA_OVERRIDE}" "${CN_SCHEMA_DIR}"
+	#sed -i "s|@CN_BROWSER@|${CN_BROWSER}|g" "${CN_SCHEMA_OVERRIDE}"
+	#cp "${CN_SCHEMA_OVERRIDE}" "${CN_SCHEMA_DIR}"
 	glib-compile-schemas "${CN_SCHEMA_DIR}"
 }
 
@@ -72,10 +72,10 @@
 	set_gsettings
 
 	# Set gdm shell logo
-	cp /usr/share/antergos/logo.png ${CN_DESTDIR}/usr/share/antergos/
+	#cp /usr/share/antergos/logo.png ${CN_DESTDIR}/usr/share/antergos/
 
 	# Set skel directory
-	cp -R ${CN_DESTDIR}/home/${CN_USER_NAME}/.config ${CN_DESTDIR}/etc/skel
+	#cp -R ${CN_DESTDIR}/home/${CN_USER_NAME}/.config ${CN_DESTDIR}/etc/skel
 
 	# xscreensaver config
 	set_xscreensaver
@@ -210,34 +210,34 @@
 
 mate_settings() {
 	# Set MATE in .dmrc
-	echo "[Desktop]" > "${CN_DESTDIR}/home/${CN_USER_NAME}/.dmrc"
-	echo "Session=mate-session" >> "${CN_DESTDIR}/home/${CN_USER_NAME}/.dmrc"
+	#echo "[Desktop]" > "${CN_DESTDIR}/home/${CN_USER_NAME}/.dmrc"
+	#echo "Session=mate-session" >> "${CN_DESTDIR}/home/${CN_USER_NAME}/.dmrc"
 
 	# Set gsettings
 	set_gsettings
 
 	# Set MintMenu Favorites
-	if [[ "${CN_BROWSER}" = 'firefox' ]]; then
-		sed -i 's|chromium|firefox|g' /usr/share/cnchi/scripts/postinstall/applications.list
-	fi
+	#if [[ "${CN_BROWSER}" = 'firefox' ]]; then
+	#	sed -i 's|chromium|firefox|g' /usr/share/cnchi/scripts/postinstall/applications.list
+	#fi
 
-	cp /usr/share/cnchi/scripts/postinstall/applications.list "${CN_DESTDIR}/usr/lib/linuxmint/mintMenu/applications.list"
+	#cp /usr/share/cnchi/scripts/postinstall/applications.list "${CN_DESTDIR}/usr/lib/linuxmint/mintMenu/applications.list"
 
 	# Copy panel layout and make it the default
-	cd "${CN_DESTDIR}/usr/share/mate-panel/layouts"
-	cp /usr/share/cnchi/scripts/antergos.layout .
-	rm default.layout
-	ln -sr antergos.layout default.layout
-	cd -
+	#cd "${CN_DESTDIR}/usr/share/mate-panel/layouts"
+	#cp /usr/share/cnchi/scripts/antergos.layout .
+	#rm default.layout
+	#ln -sr antergos.layout default.layout
+	#cd -
 
 	# Work-around for bug in mate-panel
-	CN_POST_INSTALL_DIR=/usr/share/cnchi/scripts/postinstall
-	CN_HOTFIX_SCRIPT="${CN_POST_INSTALL_DIR}/first-boot-hotfix.sh"
-	CN_HOTFIX_DESKTOP="${CN_POST_INSTALL_DIR}/first-boot-hotfix.desktop"
-	cp "${CN_HOTFIX_SCRIPT}" "${CN_DESTDIR}/usr/bin"
-	mkdir -p "${CN_DESTDIR}/home/${CN_USER_NAME}/.config/autostart"
-	cp "${CN_HOTFIX_DESKTOP}" "${CN_DESTDIR}/home/${CN_USER_NAME}/.config/autostart"
-	chmod +x "${CN_DESTDIR}/usr/bin/first-boot-hotfix.sh"
+	#CN_POST_INSTALL_DIR=/usr/share/cnchi/scripts/postinstall
+	#CN_HOTFIX_SCRIPT="${CN_POST_INSTALL_DIR}/first-boot-hotfix.sh"
+	#CN_HOTFIX_DESKTOP="${CN_POST_INSTALL_DIR}/first-boot-hotfix.desktop"
+	#cp "${CN_HOTFIX_SCRIPT}" "${CN_DESTDIR}/usr/bin"
+	#mkdir -p "${CN_DESTDIR}/home/${CN_USER_NAME}/.config/autostart"
+	#cp "${CN_HOTFIX_DESKTOP}" "${CN_DESTDIR}/home/${CN_USER_NAME}/.config/autostart"
+	#chmod +x "${CN_DESTDIR}/usr/bin/first-boot-hotfix.sh"
 }
 
 nox_settings() {
