--- PKGBUILD.orig	2017-02-16 01:54:16.038854831 +0300
+++ PKGBUILD	2017-02-16 01:58:11.298856844 +0300
@@ -10,9 +10,9 @@
 arch=('i686' 'x86_64')
 url="http://www.gegl.org"
 license=('GPL3' 'LGPL3')
-depends=('babl>=0.1.24' 'libspiro' 'json-glib')
+depends=('babl>=0.1.24' 'json-glib')
 makedepends=('git' 'intltool' 'python2' 'ruby' 'lua'
-             'libraw' 'openexr' 'ffmpeg' 'librsvg' 'jasper'
+             'libraw' 'openexr' 'ffmpeg' 'librsvg'
              'libtiff' 'suitesparse' 'gobject-introspection')
 optdepends=('openexr: for using the openexr plugin'
             'ffmpeg: for using the ffmpeg plugin'
@@ -35,8 +35,7 @@
   cd "$srcdir/$_gitname"
 
   ./autogen.sh
-  ./configure --prefix=/usr --with-sdl --with-openexr --with-librsvg \
-    --with-libavformat --with-jasper --disable-docs \
+./configure --prefix=/usr --disable-docs --enable-debug=0 --without-jasper --without-exiv2 --without-sdl --without-libavformat --without-libv4l --without-libspiro --without-graphviz \
     --enable-workshop \
 	--enable-introspection=yes
   make
